spec/yaml/swiftnav/sbp/profiling.yaml

```diff
@@ -52,3 +52,120 @@ definitions:
+      sender to sender. Refer to product documentation to understand the exact scope
+      and meaning of this message.
+    fields:
+      - total_cpu_time:
+          type: u64
+          desc: Total cpu time in microseconds consumed by this system
+      - age:
+          type: u64
+          desc: Age of the producing system in microseconds
+      - n_threads:
+          type: u8
+          desc: Number of threads being tracked by this system
+      - heap_usage:
+          type: u32
+          desc: Number of bytes allocated on the heap
+
+ - MSG_PROFILING_THREAD_INFO:
+    id: 0xCF02
+    short_desc: Thread Profiling Information
+    desc: >
+      Contains profiling information related to a single thread being tracked by
+      the producing system. Refer to product documentation to understand the exact scope
+      and meaning of this message.
+    fields:
+      - total_cpu_time:
+          type: u64
+          desc: Total cpu time in microseconds consumed by this thread
+      - age:
+          type: u64
+          desc: Age of the thread in microseconds
+      - state:
+          type: u8
+          desc: Thread state
+          fields:
+            - 0-1:
+                desc: Thread state
+                values:
+                  - 0: External
+                  - 1: Running
+                  - 2: Stopped
+                  - 3: Reserved
+      - stack_size:
+          type: u32
+          desc: Stack size in bytes
+      - stack_usage:
+          type: u32
+          desc: Stack highwater usage in bytes
+      - name:
+          type: string
+          encoding: null_terminated
+          desc: Thread name
+
+ - ResourceBucket:
+    short_desc: A bucket containing various resources
+    desc: >
+      Information about allocation of various resources grouped by buckets. Refer to product
+      documentation to understand the meaning and values in this message.
+    fields:
+      - name:
+          type: string 
+          encoding: null_terminated
+          size: 21
+          desc: Bucket name
+      - thread:
+          type: u8
+          desc: Number of threads
+      - mutex:
+          type: u8
+          desc: Number of mutexes
+      - cv:
+          type: u8
+          desc: Number of condition variables
+      - io:
+          type: u8
+          desc: Number of IO handles
+      - heap_bytes_alloc:
+          type: u32
+          desc: Number of bytes allocated on the heap
+      - heap_bytes_free:
+          type: u32
+          desc: Number of bytes freed on the heap
+      - io_write:
+          type: u32
+          desc: Number of bytes written to IO handles
+      - io_read:
+          type: u32
+          desc: Number of bytes read from IO handles
+          
+
+ - MSG_PROFILING_RESOURCE_COUNTER:
+    id: 0xCF03
+    short_desc: Information about resource buckets
+    desc: >
+      Information about resource buckets. Refer to product documentation to understand
+      the meaning and values in this message.
```

Comment:
is this description correct? I guess it contains buckets but not exactly what I was expecting give the message name